.github/workflows/terraform-apply.yaml

```
@@ -0,0 +1,57 @@
+# Based on https://github.com/canonical/sunbeam-terraform/blob/9973905d24892903fab19ce691bbc5c28c53df0a/.github/workflows/deploy.yml
+# For usage, refer to templates/terraform-deploy-usage-template.yaml
+# Applies terraform module and wait for the application to go active or blocked.
+name: Apply Terraform module
+
+on:
+  workflow_call:
+    inputs:
+      model:
+        description: The model that the charm is deployed on. Defaults to `testing`.
+        required: false
+        type: string
+        default: testing
+      channel:
+        description: The channel that the charm is deployed from. Defaults to `latest/stable`.
+        required: false
+        type: string
+        default: latest/stable
+      module-directory:
+        description: The directory that stores the Terraform module. Defaults to `./terraform`.
+        required: false
+        type: string
+        default: ./terraform
+
+jobs:
+  apply:
+    name: Apply Terraform
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+        with:
+          provider: microk8s
+          channel: 1.29-strict/stable
+          microk8s-addons: "dns storage rbac metallb:10.64.140.43-10.64.140.49"
```

I'd say that 90% of our charms won't need the metallb addon, only the ones that integrate with istio and have tests that use it, otherwise we've seen this slows down deployments and uses a lot of resources.
I wonder if we should remove this step entirely and leave it to the repo's workflow to call this action to setup the env and then the workflow (I can't remember if this is problematic).

Also,
```suggestion
          microk8s-addons: "dns hostpath-storage rbac metallb:10.64.140.43-10.64.140.49"
```

---

Addressed